### PR TITLE
[KERNEL32_APITEST] Fix and improve InitOnce test

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fixes stack corruption on x64 and adds an additional test case.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]

File: modules/rostests/apitests/kernel32/InitOnce.c [hunk 1]
```diff
@@ -169,6 +170,16 @@ START_TEST(InitOnce)
         goto _test_async;
     }
 
+    /* Call again to check whether initialization has completed */
+    fPending = 0xdeadbeef;
+    bRet = pfnInitOnceBeginInitialize(&InitOnce,
+                                      INIT_ONCE_CHECK_ONLY,
+                                      &fPending,
+                                      (LPVOID*)&ulContextData);
+    ok(bRet == FALSE, "InitOnceBeginInitialize should fail\n");
```
Comment (katahiromz) [inline modules/rostests/apitests/kernel32/InitOnce.c:179]:
```suggestion
    ok(!bRet, "InitOnceBeginInitialize should fail\n");
```

Comment (tkreuzer) [reply modules/rostests/apitests/kernel32/InitOnce.c:179]: You are nitpicking.

Comment (JoachimHenze) [reply modules/rostests/apitests/kernel32/InitOnce.c:179]: Sure he does. But "short is good"(TM). I see nothing wrong with aiming for excellence in that regard.

File: modules/rostests/apitests/kernel32/InitOnce.c [hunk 2]
```diff
@@ -154,6 +154,7 @@ START_TEST(InitOnce)
 
     /* Re-initialize One-Time initialization structure by using INIT_ONCE_STATIC_INIT */
     InitOnce = (INIT_ONCE)INIT_ONCE_STATIC_INIT;
+    ulContextData = 0xdeadbeef;
```
Comment (HBelusca) [inline modules/rostests/apitests/kernel32/InitOnce.c:157]: What about using `0xdeadbeefdeadbeefULL` like done in other apitests with x64 support?

**Review by HBelusca — APPROVED** [review]

File: modules/rostests/apitests/kernel32/InitOnce.c (hunk 1 above)
Comment (tkreuzer) [reply modules/rostests/apitests/kernel32/InitOnce.c:179]: No, that is complete nonsense.

File: modules/rostests/apitests/kernel32/InitOnce.c (hunk 2 above)
Comment (tkreuzer) [reply modules/rostests/apitests/kernel32/InitOnce.c:157]: it doesn't really matter.

**Pull request merged**